account_payment_partner/models/account_move.py

```diff
@@ -45,7 +45,7 @@ def _compute_payment_mode_filter_type_domain(self):
 
     @api.depends("partner_id", "move_type")
     def _compute_partner_bank_filter_type_domain(self):
-        for move in self:
+        for move in self.filtered(lambda m: not m.partner_bank_id):
```

Comment:
This is not correct, as all fields must have a result or there will be cache miss. You can do `self.partner_bank_filter_type_domain = self.partner_bank_filter_type_domain` for avoiding, but functionally I haven't checked what you want to achieve. Anyways, it sounds weird that you are acting on the domain.